File: blog/20101213/func_compose/compose_sample.scala

val plus = (x: Int) => x + 3
val times = (x: Int) => x * 2

val f = plus andThen times
val g = plus compose times

// times(plus(4)) = 14
println(f(4))
// plus(times(4)) = 11
println(g(4))

File: blog/20130901/java8_andthen_compose/ComposeSample.java
import java.util.function.Function;

public class ComposeSample {
	public static void main(String... args) {
		Function<Integer, Integer> plus = (x) -> x + 3;
		Function<Integer, Integer> times = (x) -> x * 2;

		// (4 + 3) * 2 = 14 : plus -> times
		System.out.println(plus.andThen(times).apply(4));
		// (4 * 2) + 3 = 11 : times -> plus
		System.out.println(plus.compose(times).apply(4));
	}
}